runtime/src/bucket_map_holder.rs

```diff
@@ -53,6 +59,12 @@ impl<T: IndexValue> BucketMapHolder<T> {
     pub fn future_age_to_flush(&self) -> Age {
         self.current_age().wrapping_add(self.ages_to_stay_in_cache)
     }
+
+    fn has_age_interval_elapsed(&self) -> bool {
+        // note that when this returns true, state of age_timer is modified
+        self.age_timer.should_update(AGE_MS) // todo: make sure this interval has correct Ordering
```

Comment:
Should this todo be resolved before the PR gets merged?

Reply:
I was hoping you'd help me resolve this. I added tests and it appears to work. should_update uses Ordering::Relaxed.

Reply:
I got rid of the comment for now. Please do look at AtomicInterval and see if the orderings make sense.

Reply:
Gotcha, I'll take a look

Reply:
I think this looks good